src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -0,0 +1,110 @@
+        return (BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor) super.getDescriptor();
+    }
+
+    @Override
+    public void start(MultiBranchProject<?, ?> project, boolean newInstance) {
+        super.start(project, newInstance);
+        if (!newInstance) {
+            return;
+        }
+        BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor descriptor = getDescriptor();
+        project.getSCMSources().stream().filter(scm -> scm instanceof BitbucketSCMSource)
+                .filter(scm -> ((BitbucketSCMSource) scm).isValid())
+                .forEach(scm -> {
+            boolean isAdded = descriptor.addTrigger(project, (BitbucketSCMSource) scm);
+            ((BitbucketSCMSource) scm).setWebhookRegistered(isAdded);
+        });
+    }
+
+    @Symbol("BitbucketWebhookMultibranchTriggerImpl")
+    @Extension
+    public static class BitbucketWebhookMultibranchTriggerDescriptor extends TriggerDescriptor {
+
+        @Inject
+        private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+        @Inject
+        private RetryingWebhookHandler retryingWebhookHandler;
+
+        @SuppressWarnings("unused")
+        public BitbucketWebhookMultibranchTriggerDescriptor() {
+
+        }
+
+        public BitbucketWebhookMultibranchTriggerDescriptor(RetryingWebhookHandler webhookHandler,
+                                                            BitbucketPluginConfiguration bitbucketPluginConfiguration) {
+            this.retryingWebhookHandler = webhookHandler;
+            this.bitbucketPluginConfiguration = bitbucketPluginConfiguration;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return Messages.BitbucketWebhookMultibranchTrigger_displayname();
+        }
+
+        @Override
+        public boolean isApplicable(Item item) {
+            return item instanceof MultiBranchProject;
+        }
+
+        boolean addTrigger(Item item, BitbucketSCMSource scm) {
+            try {
+                registerWebhook(item, scm.getBitbucketSCMRepository());
+                return true;
+            } catch (Exception ex) {
+                LOGGER.log(Level.SEVERE, "There was a problem while trying to add webhook", ex);
+                throw ex;
+            }
+        }
+
+        private BitbucketServerConfiguration getServer(String serverId) {
+            return bitbucketPluginConfiguration
+                    .getServerById(serverId)
+                    .orElseThrow(() -> new BitbucketClientException(
+                            "Server config not found for input server id" + serverId));
+        }
+
+        private void registerWebhook(Item item, BitbucketSCMRepository repository) {
+            requireNonNull(repository.getServerId());
+            BitbucketServerConfiguration bitbucketServerConfiguration = getServer(repository.getServerId());
```

Comment:
```suggestion
            BitbucketServerConfiguration bitbucketServerConfiguration = getServer(requireNonNull(repository.getServerId(), "repository.serverId"));
```

Reply:
Hmm actually I think we want one or two tests covering the start method here- will add those.
I'm not sure if this is a rule but I feel like we tend to separate out descriptor and decribable classes when we have a lot of tests and there's a separation in terms of functionality (SCM tests are a good example of this, as the descriptors are mostly just verifying correct data). I think in this case, given they mirror functionality so closely, it makes sense to keep the tests together under the parent class name.